templates/server/src/main/resources/archetype-resources/core/src/main/java/__packageInPathFormat__/general/common/impl/security/BaseUserDetailsService.java

```diff
@@ -60,33 +55,31 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
    * @return the associated {@link GrantedAuthority}s
    * @throws AuthenticationException if no principal is retrievable for the given {@code username}
    */
-  protected Set<GrantedAuthority> getAuthorities(String username) throws AuthenticationException {
+  protected Set<GrantedAuthority> getAuthorities(UserDetails user) throws AuthenticationException {
 
-    Objects.requireNonNull(username, "username");
     // determine granted authorities for spring-security...
-    Set<GrantedAuthority> authorities = new HashSet<>();
-    Collection<String> accessControlIds = getRoles(username);
     Set<AccessControl> accessControlSet = new HashSet<>();
-    for (String id : accessControlIds) {
-      boolean success = this.accessControlProvider.collectAccessControls(id, accessControlSet);
-      if (!success) {
-        LOG.warn("Undefined access control {}.", id);
-      }
-    }
-    for (AccessControl accessControl : accessControlSet) {
-      authorities.add(new AccessControlGrantedAuthority(accessControl));
-    }
-    return authorities;
+
+    Set<String> undefinedIds = getRoles(user).stream()
+        .filter(id -> !this.accessControlProvider.collectAccessControls(id, accessControlSet))
+        .collect(Collectors.toUnmodifiableSet());
+
+    undefinedIds.forEach(id -> LOG.warn("Undefined access control {}.", id));
+
+    return accessControlSet.stream().map(accessControl -> new AccessControlGrantedAuthority(accessControl))
+        .collect(Collectors.toUnmodifiableSet());
   }
 
-  private Collection<String> getRoles(String username) {
+  static final String ROLE_PREFIX = "ROLE_";
```

Comment:
We could find a better spot making this constant public but already fine if it does not cause redundancies.